catalog/lib/theme/app_theme.dart

```diff
@@ -7,19 +8,41 @@ import 'package:catalog/theme/app_dimensions.dart';
 late AppColors _colors;
 late AppDimens _dimensions;
 late AppTextStyles _styles;
+late AppButtons _buttons;
 
 class AppTheme {
   static ThemeData provideAppTheme(BuildContext buildContext) {
     // It can be changed based on the device
     _styles = AppTextStyles.getAppStyles();
     _dimensions = AppDimens.getDimensions();
     _colors = AppColors.getColorScheme();
+    _buttons = AppButtons.getButtonTheme();
 
     return ThemeData(
       primaryColor: _colors.primary,
       colorScheme: _colors,
-      textTheme: _styles.getThemeData(),
-      primaryTextTheme: _styles.getThemeData(),
+      filledButtonTheme: FilledButtonThemeData(
+        style: _buttons.filledButton,
+      ),
+      outlinedButtonTheme: OutlinedButtonThemeData(
+        style: _buttons.outlineButton,
+      ),
+      textButtonTheme: TextButtonThemeData(
+        style: _buttons.textButton,
+      ),
+      scaffoldBackgroundColor: _colors.primary.shade200,
+      textTheme: _styles.getThemeData().apply(
+        bodyColor: _colors.textColor.shade100,
+        displayColor: _colors.textColor.shade500,
+      ),
+      primaryTextTheme: _styles.getThemeData().apply(
+        bodyColor: _colors.textColor.shade100,
+        displayColor: _colors.primary,
```

Comment:
Maybe use white for displayColors instead of the current primary?, since this property is used in contrast to the primary colors